include/linux/double-fetch-detection.h

```diff
@@ -0,0 +1,17 @@
+#ifndef __LINUX_DOUBLE_FETCH_DETECTION_H__
+#define __LINUX_DOUBLE_FETCH_DETECTION_H__
+#ifdef CONFIG_DF_DETECTION
+
+struct address{
+    void *startaddress;
+    unsigned long len;
```

Comment:
why not 'size_t'?

Reply:
as far as i know size_t is unsigned long ?  is defined in include/linux/types.h as unsigned long. 
i wrote it unsigned long just like the datatype of the parameter (n) in  copy_from_user function

Reply:
If you want consistency with the copy..user interface, 'unsigned long' is fine. But I also do not see why we can't just use size_t here. I'd probably use size_t, just because it's a little more precise and there should be no harm in doing so.